kubernetes rack awareness (#904)
redpanda-data/documentation: modules/reference/partials/enterprise-licensed-property.adoc

> NOTE: This property requires a license when used with certain values.  Check the field *Enterprise license required*. For license details, see xref:get-started:licenses.adoc[].

Reviewer: my first question as a customer would be, what do you mean by "certain values"? The second sentence, "Check the field *Enterprise license required*."...Check it for what, exactly?

Reviewer 2: take "sasl_mechanism" for example. Enterprise is not required when using `SCRAM` but it's required when using `GSSAPI` or `OAUTHBEARER`

Reviewer 2: maybe ?
```suggestion
NOTE: Some values for this property require an Enterprise license. Refer to the *Enterprise license required* field for specific requirements. For license details, see xref:get-started:licenses.adoc[].
```

Reviewer: LGTM!
